homeassistant/components/cover/opengarage.py

```diff
@@ -0,0 +1,242 @@
+            CONF_DEVICEKEY: device_config.get(CONF_DEVICEKEY)
+        }
+
+        covers.append(OpenGarageCover(hass, args))
+
+    add_devices(covers)
+
+
+class OpenGarageCover(CoverDevice):
+    """Representation of a OpenGarage cover."""
+
+    # pylint: disable=no-self-use
+    def __init__(self, hass, args):
+        """Initialize the cover."""
+        self.opengarage_url = 'http://{}:{}'.format(
+            args[CONF_HOST],
+            args[CONF_PORT])
+        self.hass = hass
+        self._name = args[CONF_NAME]
+        self.device_id = args['device_id']
+        self._devicekey = args[CONF_DEVICEKEY]
+        self._state = STATE_UNKNOWN
+        self._state_before_move = None
+        self.dist = None
+        self.signal = None
+        self._unsub_listener_cover = None
+        self._available = True
+
+        # Lets try to get the configured name if not provided.
+        try:
+            if self._name is None:
+                doorconfig = self._get_status()
+                if doorconfig["name"] is not None:
+                    self._name = doorconfig["name"]
+            self.update()
+        except requests.exceptions.ConnectionError as ex:
+            _LOGGER.error('Unable to connect to server: %(reason)s',
+                          dict(reason=ex))
+            self._state = STATE_OFFLINE
+            self._available = False
+            self._name = DEFAULT_NAME
+        except KeyError as ex:
+            _LOGGER.warning('OpenGarage device %(device)s seems to be offline',
+                            dict(device=self.device_id))
+            self._name = DEFAULT_NAME
+            self._state = STATE_OFFLINE
+            self._available = False
+
+    @property
+    def name(self):
+        """Return the name of the cover."""
+        return self._name
+
+    @property
+    def should_poll(self):
+        """No polling needed for a demo cover."""
+        return True
+
+    @property
+    def available(self):
+        """Return True if entity is available."""
+        return self._available
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        data = {}
+
+        if self.signal is not None:
+            data[ATTR_SIGNAL_STRENGTH] = self.signal
+
+        if self.dist is not None:
+            data[ATTR_DISTANCE_SENSOR] = self.dist
+
+        if self._state is not None:
+            data[ATTR_DOOR_STATE] = self._state
+
+        return data
+
+    @property
+    def is_closed(self):
+        """Return if the cover is closed."""
+        if self._state == STATE_UNKNOWN:
+            return None
+        else:
+            return self._state in [STATE_CLOSED, STATE_OPENING]
+
+    def _start_watcher(self, command):
+        """Start watcher."""
+        _LOGGER.debug("Starting Watcher for command: %s ", command)
+        if self._unsub_listener_cover is None:
+            self._unsub_listener_cover = track_utc_time_change(
```

Comment:
It seems like you want to keep calling update while the state is in opening/closing. Why not have this logic inside the update method ? When update is done, if certain state, schedule another sync.

Also use `track_point_in_time` so that you don't have to unsub anything.